src/services/env.ts

```diff
@@ -0,0 +1,73 @@
+import { PATH_CONFIG_DEFAULT } from '@/pathConfigDefault'
+import { PathConfig } from '@/types/index'
+
+export type EnvVars = {
+  KUMA_NAME: string
+  KUMA_VERSION: string
+  KUMA_BASE_PATH: string
+  KUMA_API_URL: string
+  KUMA_DOCS_URL: string
+  KUMA_UTM_QUERY_PARAMS: string
+  KUMA_FEEDBACK_URL: string
+  KUMA_CHAT_URL: string
+}
+export default class Env {
+  env: EnvVars
+  constructor(env: EnvVars) {
+    this.env = env
+  }
+
+  var(key: keyof EnvVars) {
+    return this.env[key]
+  }
+}
+
+// TODO(jc): for ease just do all this here for until we have a proper container
+const config = readPathConfigFromDom()
+const version = semver(config.version)
+
+export const env = new Env({
+  KUMA_NAME: `${import.meta.env.VITE_NAMESPACE}`,
+  KUMA_VERSION: version.pre,
+  KUMA_API_URL: config.apiUrl,
+  KUMA_BASE_PATH: config.baseGuiPath,
+  // TODO(jc): not totally sure we need to use a regex here, maybe just split and join if not
+  KUMA_DOCS_URL: `${import.meta.env.VITE_DOCS_BASE_URL}/${version.patch === '0.0.0' ? 'dev' : version.patch.replace(/\.\d+$/, '.x')}`,
+  // remove the ? for now incase we need to append to an already query
+  // param'ed URL
+  KUMA_UTM_QUERY_PARAMS: import.meta.env.VITE_UTM.substring(1),
+  KUMA_FEEDBACK_URL: import.meta.env.VITE_FEEDBACK_URL,
+  KUMA_CHAT_URL: import.meta.env.VITE_CHAT_URL,
+})
```

Comment:
**Suggetion**: The sole purpose of the `env` utility is to look up static primitive values. This shouldn’t be based on a class whose only functionality is to look up object values by property.

```suggestion
export const env: EnvVars = {
  KUMA_NAME: `${import.meta.env.VITE_NAMESPACE}`,
  KUMA_VERSION: version.pre,
  KUMA_API_URL: config.apiUrl,
  KUMA_BASE_PATH: config.baseGuiPath,
  // TODO(jc): not totally sure we need to use a regex here, maybe just split and join if not
  KUMA_DOCS_URL: `${import.meta.env.VITE_DOCS_BASE_URL}/${version.patch === '0.0.0' ? 'dev' : version.patch.replace(/\.\d+$/, '.x')}`,
  // remove the ? for now incase we need to append to an already query
  // param'ed URL
  KUMA_UTM_QUERY_PARAMS: import.meta.env.VITE_UTM.substring(1),
  KUMA_FEEDBACK_URL: import.meta.env.VITE_FEEDBACK_URL,
  KUMA_CHAT_URL: import.meta.env.VITE_CHAT_URL,
}
```

Reply:
I can see your point here, right now, but moving forwards all Services are probably going to be classes or maybe functions, not bare objects.

Reply:
I’m not a fan of abstracting away simple utilities into more complex utilities for the sake of having all utilities follow the same pattern (i.e. a service pattern).

Why do we want this?

None of the properties in this object need any sort of additional handling or routine. It’s merely looking up one primitive value for each use. So while perhaps some utilities/application services are more complex in nature, looking up static data isn’t. That should remain a simple task and easy maintenance even if other services are used with said pattern.

Reply:
> Why do we want this?

Because we want to use a consistent pattern to inject dependencies from The Outside without having to change code. Granted this is a simple service that could be done differently, but because its simple, it's a good opportunity to get the ball rolling.

Reply:
What about adding computed properties? For example clearly the UTM stuff can be computed.

Reply:
We had a quick chat about this. We can calculate the utm from the `VITE_NAMESPACE`/`KUMA_NAME`, which means we can drop `VITE_UTM` entirely, gonna add a commit on the end here for that.